src/main/java/org/radarbase/management/config/SecurityConfiguration.kt

```diff
@@ -0,0 +1,105 @@
+package org.radarbase.management.config
+
+import org.radarbase.management.security.Http401UnauthorizedEntryPoint
+import org.radarbase.management.security.RadarAuthenticationProvider
+import org.springframework.beans.factory.BeanInitializationException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpMethod
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.DefaultAuthenticationEventPublisher
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.builders.WebSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.data.repository.query.SecurityEvaluationContextExtension
+import org.springframework.security.web.authentication.logout.LogoutSuccessHandler
+import tech.jhipster.security.AjaxLogoutSuccessHandler
+import javax.annotation.PostConstruct
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+open class SecurityConfiguration
```

Comment:
Btw if you end up with cyclic dependencies using constructor autowiring, you could consider autowiring a `Provider<MyService>` instead of the service itself. This breaks the dependency cycle as long as you don’t invoke the provider in the constructor.